hm_pyhelper/diagnostics/diagnostics_report.py

```diff
@@ -16,6 +16,16 @@ class DiagnosticsReport(dict):
         'public_key', 'bluetooth', 'VARIANT', 'FREQ', 'serial_number'
     }
 
+    KEYS_TO_CHECK_IN_GATEWAY_TRANSACTION_PAYLOAD = {
+        'address', 'fee', 'owner', 'payer', 'txn', 'staking fee'
+    }
+
+    ADD_GATEWAY_TXN_NAME_KEY = 'destination_name'
+    ADD_GATEWAY_TXN_PAYLOAD_KEY = 'destination_add_gateway_txn'
```

Comment:
Also, updating hm-pyhelper is a pain. You have to cut a new release before using it in HPT or elsewhere. Any advantage of centralizing acceptance logic here, seems offset by the extra lift it takes to propagate changes into client code.

Reply:
> Also, updating hm-pyhelper is a pain. You have to cut a new release before using it in HPT or elsewhere. Any advantage of centralizing acceptance logic here, seems offset by the extra lift it takes to propagate changes into client code.

I agree with that. The reason this validation was done in DiagnosticsReport was because we were already validating for errors and manufacturing errors. So instead of having validation logic at two different repos it was better to have it in one. However I agree with you that all such logic is better done in HPT.

On the other hand what advantages do we get with the DiagnosticsReport class instead of plain dict in that case? In HPT `Miner` class we have `payload` and `diagnostics_report`. Currently the only reason that I can think of for having diagnostics_report in there was this validation. Otherwise we can just do the validations on the payload itself.

Use of DiagnosticsReport in hm-diag makes sense but without validation logic it isn't very useful in HPT.

Reply:
@marvinmarnold  May be shared code is better off as a git submodule rather than a separate package?

Reply:
> Use of DiagnosticsReport in hm-diag makes sense but without validation logic it isn't very useful in HPT.

That's fair. DiagnosticsReport does not add a ton of value in HPT now. When DiagnosticsReport was originally written, we were still using the concept of `PF` which had some distinct logic, but we've since moved away from that. To take the argument further, maybe Diagnostics and Diagnostics report should never have been included in pyhelper and belong in hm-diag. But then HPT would need to either treat the report as a dict as you suggest or import hm-diag.

> The reason this validation was done in DiagnosticsReport was because we were already validating for errors and manufacturing errors.

To be clear, I don't think `has_manufacturing_errors` belongs in DiagnosticsReport either. I think it should be `has_errors`. Then HPT would invoke that and supply the errors it cares about. If I recall correctly, we decided on `has_manufacturing_errors` because in practice the only use case we have today is manufacturing. But now that decision has led to an anti-pattern in my opinion.

Reply:
> But then HPT would need to either treat the report as a dict as you suggest or import hm-diag.
Yes so far at HPT side just using the `payload` instead of `DiagnosticsReport` seems simpler and we don't miss anything. All validation logic could have been part of HPT easily. With diagnostics report I assumed that errors related logic would get used in other places like hm-diag.

> If I recall correctly, we decided on has_manufacturing_errors because in practice the only use case we have today is manufacturing. But now that decision has led to an anti-pattern in my opinion.
Yes we're on the same page on this. We get a dict then make it into a diagnostics report and validate that. So we move validation out there is a much simpler path of validating against the dict using validation logic in HPT itself.

Reply:
Moved manufacturing errors and transaction errors handling logic to HPT so now DiagnosticsReport is not tied to those any more. This resolves our conversation above.